tests/agent/test_agent.py

```diff
@@ -605,3 +606,21 @@ def test_correlation_id_header(
         )
         agent = GXAgent()
         agent.run()
+
+
+def test_raise_gx_cloud_err_on_http_error_error_response():
+    agent = GXAgent()
+    test_response = requests.Response()
+    # 404 - Not Found
+    test_response.status_code = 404
+    with pytest.raises(gx_exception.GXCloudError):
+        agent._raise_gx_cloud_err_on_http_error(test_response)
+
+
+def test_raise_gx_cloud_err_on_http_error_success_response():
+    agent = GXAgent()
+    test_response = requests.Response()
+    # 200 - OK
+    test_response.status_code = 200
+    # no Exception raised
+    agent._raise_gx_cloud_err_on_http_error(test_response)
```

Comment:
```suggestion
def test_raise_gx_cloud_err_on_http_error_error_response():
    test_response = requests.Response()
    # 404 - Not Found
    test_response.status_code = 404
    with pytest.raises(gx_exception.GXCloudError):
        GXAgent._raise_gx_cloud_err_on_http_error(test_response)


def test_raise_gx_cloud_err_on_http_error_success_response():
    test_response = requests.Response()
    # 200 - OK
    test_response.status_code = 200
    # no Exception raised
    GxAgent._raise_gx_cloud_err_on_http_error(test_response)
```
You shouldn't need to create an instance of the agent to test this method.
I would strongly suggest making this a class or static method.